packages/js/product-editor/src/components/header/header.tsx

```diff
@@ -1,14 +1,76 @@
 /**
  * External dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { Product } from '@woocommerce/data';
+import { Button } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { createElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { navigateTo, getNewPath } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import { AUTO_DRAFT_NAME } from '../../utils';
 
 export type HeaderProps = {
-	title: string;
+	productId: number;
+	productName: string;
 };
 
-export function Header( { title }: HeaderProps ) {
+export function Header( { productId, productName }: HeaderProps ) {
+	const { isProductLocked, isSaving, product } = useSelect(
+		( select ) => {
+			const { isSavingEntityRecord, getEditedEntityRecord } =
+				select( 'core' );
+			const { isPostSavingLocked } = select( 'core/editor' );
+			return {
+				isProductLocked: isPostSavingLocked(),
+				isSaving: isSavingEntityRecord(
+					'postType',
+					'product',
+					productId
+				),
+				product: getEditedEntityRecord(
+					'postType',
+					'product',
+					productId
+				) as Product,
+			};
+		},
+		[ productId ]
+	);
+
+	const isDisabled = isProductLocked || isSaving;
+	const isProductNameNotEmpty = Boolean( product?.name );
+	const isProductNameDirty = product.name !== productName;
+	const isCreating = productName === AUTO_DRAFT_NAME;
+
+	let title = '';
```

Comment:
And what is the name you suggest here for that util function? We already have `get-product-title`. It would be great if we could reuse the same one but it has a different behavior. Do you like this one `get-header-title` for the new function?

Reply:
done -> [1d0165e](https://github.com/woocommerce/woocommerce/pull/37152/commits/1d0165e875634df4d66ba20d715574f9e5423d29)

I did not add tests to that function now because testing are not configured for the current product-editor package yet. That configuration is been manage in other PR.